base/gmp.jl

```diff
@@ -42,9 +42,16 @@ mutable struct BigInt <: Integer
     alloc::Cint
     size::Cint
     d::Ptr{Limb}
+
     function BigInt()
-        b = new(zero(Cint), zero(Cint), C_NULL)
-        MPZ.init!(b)
+        b = MPZ.init!(new())
+        finalizer(b, cglobal((:__gmpz_clear, :libgmp)))
+        return b
+    end
+
+    # n is the number of bits to allocate
+    function BigInt(::typeof(sizehint!), n)
```

Comment:
keyword arguments are likely to get faster fairly soon, so we should avoid making too many ugly API choices as workarounds for them being slow now